database/tbcd/level/level.go

```diff
@@ -1486,6 +1486,28 @@ func (l *ldb) BlockByHash(ctx context.Context, hash *chainhash.Hash) (*btcutil.B
 	return b, nil
 }
 
+func (l *ldb) BlockExistsByHash(ctx context.Context, hash chainhash.Hash) (bool, error) {
+	log.Tracef("BlockExistsByHash")
+	defer log.Tracef("BlockExistsByHash exit")
+
+	if l.cfg.blockCacheSize > 0 {
+		// Try cache first
+		if _, ok := l.blockCache.Get(hash); ok {
+			return true, nil
+		}
+	}
+
+	bDB := l.rawPool[level.BlocksDB]
+	ok, err := bDB.Has(hash[:])
+	if err != nil {
+		if errors.Is(err, leveldb.ErrNotFound) {
+			return false, nil
+		}
+		return false, fmt.Errorf("block exists: %w", err)
```

Comment:
"block exists:" makes it sound like the block existing is the error, which isn't the case here - could we make this clearer:
```suggestion
		return false, fmt.Errorf("check block exists: %w", err)
```